database/interface.py

```diff
@@ -503,6 +402,15 @@ def update_harvest_record(self, record_id, updates):
             self.db.rollback()
             return None
 
+    def delete_harvest_record(self, ckan_id):
+        records = self.db.query(HarvestRecord).filter_by(ckan_id=ckan_id).all()
```

Comment:
should there be cases where we have more than one record here?

Reply:
I noticed duplicate records with the same ckan_id in the current development environment, but I'm not sure how they ended up there.

Reply:
Should we long the count temporarily to determine if this issue persists?

Reply:
just added a log message for the duplicate ckan_id cases.